### PR TITLE
Explicitly close IPC connections

### PR DESCRIPTION
**Pull request by dwoz**
### What does this PR do?

Explicitly close IPC connections.

### Tests written?

No - Covered by existing tests

### Commits signed with GPG?

Yes

### CODE REVIEW

**Review by waynew — APPROVED** [review]

File: salt/utils/decorators/__init__.py [hunk 1]
```diff
@@ -560,6 +560,7 @@ def _decorate(*args, **kwargs):
             return self._call_function(kwargs)
 
         _decorate.__doc__ = self._function.__doc__
+        _decorate.__wrapped__ = self._function
```
Comment (s0undt3ch) [inline salt/utils/decorators/__init__.py]: `__wrapped__` is added on Py3 by `functools.wraps`, want to check for that?

**Review by s0undt3ch — CHANGES_REQUESTED** [review]
> New in version 3.2: Automatic addition of the `__wrapped__` attribute.
> New in version 3.2: Copying of the `__annotations__` attribute by default.
> Changed in version 3.4: The `__wrapped__` attribute now always refers to the wrapped function, even if that function defined a `__wrapped__` attribute. (see bpo-17482)

File: salt/utils/decorators/__init__.py (hunk 1 above)
Comment (dwoz) [reply salt/utils/decorators/__init__.py]: Nope, wrapped returns a partial. The `__wrapped__` attribute get's added when the partial is called. This adds a `__wrapped__` attribute to that partial so the test can detect it and behave appropriately.

https://github.com/python/cpython/blob/master/Lib/functools.py#L74

Comment (s0undt3ch) [reply salt/utils/decorators/__init__.py]: :+1:

**Review by s0undt3ch — APPROVED** [review]

**Pull request merged**